src/management-system-v2/components/auth-can.tsx

```diff
@@ -1,38 +1,48 @@
 'use client';
 
-import { ReactElement, ReactNode, useEffect, useMemo, FC, PropsWithChildren } from 'react';
+import { ReactElement, useEffect, useMemo, FC, PropsWithChildren } from 'react';
 import { useAbilityStore } from '@/lib/abilityStore';
 import { useSession } from 'next-auth/react';
 import { Route } from 'next';
-import { AbilityRule, ResourceActionType } from '@/lib/ability/caslAbility';
-import { PackRule } from '@casl/ability/extra';
-import { useCsrfTokenStore } from '@/lib/csrfTokenStore';
+import {
+  ResourceActionType,
+  ResourceType,
+  resources,
+  toCaslResource,
+  resourceAction,
+} from '@/lib/ability/caslAbility';
 import { usePathname, useRouter } from 'next/navigation';
-import Ability from '@/lib/ability/abilityHelper';
-
-export type AuthCanProps = {
-  action: ResourceActionType | ResourceActionType[];
-  resource: Parameters<Ability['can']>[1];
-  notLoggedIn?: ReactElement;
-  fallback?: ReactElement;
-  fallbackRedirect?: Route;
-  loading?: ReactElement;
-};
 
-// TODO: Weil client side werden evtl. sensible Daten an den Client geschickt.
-// Auf server side ändern und eigene component für client side die aber nur für
-// buttons etc. benutzt werden sollte
-export const AuthCan: FC<PropsWithChildren<AuthCanProps>> = ({
-  action,
-  resource,
-  fallback,
-  loading: loadingAuth,
-  notLoggedIn,
-  children,
-  fallbackRedirect,
-}) => {
+/** Makes at least one key of an object required */
+type OneOf<Key, Object> = Key extends keyof Object ? Partial<Object> & Pick<Object, Key> : never;
+
+type ResourceActionOneOf = OneOf<ResourceActionType, Record<ResourceActionType, true>>;
+
+type ResourceParams = Parameters<typeof toCaslResource>;
+type ResourceOneOf = OneOf<ResourceType, Record<ResourceType, ResourceParams[1] | true>>;
+
+export type AuthCanProps = ResourceActionOneOf &
+  ResourceOneOf & {
+    notLoggedIn?: ReactElement;
+    fallback?: ReactElement;
+    fallbackRedirect?: Route;
+    loading?: ReactElement;
+  };
+
+/**
+ * Component to check if a user is allowed to perform an action on a resource
+ *
+ * @example
+ * <AuthCan create Process={record} fallback={<p>Not allowed</p>}>
+ *   <button onClick={createProcess}> Create Process </button>
+ * </AuthCan>
+ * */
+export const AuthCan: FC<PropsWithChildren<AuthCanProps>> = (props) => {
+  const { fallback, loading: loadingAuth, notLoggedIn, children, fallbackRedirect } = props;
+
   const router = useRouter();
   const { status } = useSession();
+  const environmentId = useEnvironment();
```

Comment:
With #246 to get the real space id of a personal space, we have to check if this is `my` and then replace with the user id.

Reply:
Could also make that an optional flag param for `useEnvironment(true)` I guess.

Reply:
Wouldn't it be better if the useEnvironment hook handled that itself, I think there are a lot of places where it is expected that it returns the actual space id.